protobuf-format.md

```diff
@@ -0,0 +1,181 @@
+
+| CloudEvents   | protobuf |
+| ------------- | ---------------------------------------------------------------------- |
+| Boolean       | [boolean][proto-scalars] |
+| Integer       | [sint32][proto-scalars] |
+| String        | [string][proto-scalars] |
+| Binary        | [bytes][proto-scalars] |
+| URI           | [string][proto-scalars] following [RFC 3986 §4.3][rfc3986-section43]|
+| URI-reference | [string][proto-scalars] following [RFC 3986 §4.1][rfc3986-section41] |
+| Timestamp     | [Timestamp][proto-timestamp]  |
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are represented in the protobuf
+[schema][proto-schema].
+
+Required and Optional attributes are represented explicitly as protobuf fields.
+
+CloudEvent extension attributes are represented using a map construct enabling
+direct support of the CloudEvent [type system][ce-types].
+
+```proto
+map<string, CloudEventAttribute> extensions = 1;
+
+message CloudEventAttribute {
+
+    oneof attr_oneof {
+      bool ce_boolean = 1;
+      sfixed32 ce_integer = 2;
+      string ce_string = 3;
+      bytes ce_binary = 4;
+      string ce_uri = 5;
+      string ce_uri_reference = 6;
+      google.protobuf.Timestamp ce_timestamp = 7;
+    }
+}
+```
+
+NOTE: Field assignment numbers are non-normative.
+
+In this model an extension attribute's name is used as the map *key* and is
+associated with its *value* stored in the appropriately typed property.
+
+This approach allows for extensions to be represented and transported
+with no loss of *type* information.
+
+## 3. Data
+
+The specification allows for data payloads of the following types to be explicitly represented:
+
+* string
+* bytes
+* protobuf object/message
+
+```proto
+oneof data_oneof {
+
+    // Binary data
+    bytes raw_data = 2;
+
+    // String data
+    string text_data = 3;
+
+    // Protobuf Message data
+    google.protobuf.Any proto_data = 4;
+}
+```
+
+Before encoding, a protobuf serializer MUST first determine the runtime data type
+of the content. This can be determined by examining the data for invalid UTF-8
+sequences or by consulting the `datacontenttype` attribute.
+
+* Where the data is a protobuf message it MUST be stored in the `proto_data`
+property.
+* If the implementation determines that the type of the data is text, the value
+MUST be stored in the `text_data` property.
+* If the implementation determines that the type of the data is binary, the value
+MUST be stored in the `raw_data` property.
+
+## 4. Transport
+
+Transports that allow content identification MUST use the following designation:
+
+```text
+   application/cloudevents+x-protobuf
+```
+
+## 5. Examples
+
+The following code-snippets show how proto representations might be constucted asuming the availability of some convenience methods ...
+
+```java
+public static Spec.CloudEvent plainTextExample() {
+    Spec.CloudEvent.Builder ceBuilder = Spec.CloudEvent.newBuilder();
```

Comment:
You never define the enclosing CloudEvent type, only `extensions` and `data_oneof`.

Reply:
Spec.CloudEvent *is* the enclosing type, the example method returns an instance of a CloudEvent.